### PR TITLE
US026 Modify Slot/index Action Add Filtering

### PR DESCRIPTION
**Pull request by SamDotNetDev**

### CODE REVIEW

**Review by dilmurod-rozikov — APPROVED** [review]

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs [hunk 1]
```diff
@@ -20,19 +21,41 @@ public ParkingSlotController(IParkingSlotService slotService, IParkingZoneServic
 
         public IActionResult Index(int ParkingZoneId)
         {
+            var zone = _zoneService.GetById(ParkingZoneId);
             var VMs = _slotService.GetByParkingZoneId(ParkingZoneId)
                 .Select(x => new ListItemVM(x))
                 .OrderBy(x => x.Number);
-            if(!VMs.Any())
+
+            if (zone is null)
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs:28]: write this above from vms

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs [hunk 2]
```diff
@@ -20,19 +21,41 @@ public ParkingSlotController(IParkingSlotService slotService, IParkingZoneServic
...
             }
 
-            ViewData["Name"] = _zoneService.GetById(ParkingZoneId).Name;
+            ViewData["Name"] = zone.Name;
             ViewData["ParkingZoneId"] = ParkingZoneId;
             return View(VMs);
         }
 
+        [HttpPost]
+        public IActionResult Index(int ParkingZoneId, SlotCategoryEnum? category = null, bool? isSlotFree = null)
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs]:
```suggestion
        public IActionResult Index(FilterSlotVM VM)
```

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs [hunk 3]
```diff
@@ -20,19 +21,41 @@ public ParkingSlotController(IParkingSlotService slotService, IParkingZoneServic
...
+
+            if (category.HasValue)
+            {
+                slotsQuery = slotsQuery.Where(x => x.Category == category.Value);
+            }
+
+            if (isSlotFree.HasValue)
+            {
+                slotsQuery = slotsQuery.Where(x => !x.IsInUse == isSlotFree.Value);
+            }
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs]: write a separate service for filtering

File: ParkingZoneApp/Areas/Admin/Views/ParkingSlot/Index.cshtml [hunk 4]
```diff
@@ -100,4 +126,42 @@
...
+            var isSlotFree = $('#isSlotFreeFilter').val();
+
+            // AJAX request
+            $.ajax({
+                url: '@Url.Action("Index", "ParkingSlot")',
+                type: 'POST',
+                data: {
+                    ParkingZoneId: @ViewData["ParkingZoneId"],
+                    category: category,
+                    isSlotFree: isSlotFree
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Views/ParkingSlot/Index.cshtml]: assign to a single variable and then pass to action

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**